### PR TITLE
Investigate RedDeer and E4 apps

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Comment by apodhrad** [comment]
@rawagner there are lots of errors in the verification job

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
testPR

**Comment by rawagner** [comment]
@apodhrad please take a look, tests are passing

File: plugins/org.eclipse.reddeer.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -18,3 +17,7 @@ Export-Package: org.eclipse.reddeer.core.condition,
  org.eclipse.reddeer.core.reference,
  org.eclipse.reddeer.core.resolver,
  org.eclipse.reddeer.core.util
+Import-Package: org.eclipse.reddeer.workbench.core.lookup;resolution:=optional,
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.core/META-INF/MANIFEST.MF]: Is there any situation were the package is not needed? Moreover, get back the version range [2.1, 2.2)

File: plugins/org.eclipse.reddeer.core/src/org/eclipse/reddeer/core/lookup/WidgetLookup.java [hunk 2]
```diff
@@ -213,8 +214,8 @@ public Control findParent(){
 	 */
 	public boolean isExtraShellActive() {
 		Shell activeWorkbenchParentShell = null;
-		if(getWorkbenchLookup() != null){
-			activeWorkbenchParentShell = getWorkbenchLookup().getShellForActiveWorkbench();
+		if(getWorkbenchShellLookup() != null){
+			activeWorkbenchParentShell = getWorkbenchShellLookup().getWorkbenchShell();
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.core/src/org/eclipse/reddeer/core/lookup/WidgetLookup.java]: That's the reason of this change?

File: plugins/org.eclipse.reddeer.core/META-INF/MANIFEST.MF (hunk 1 above)
Comment (rawagner) [reply plugins/org.eclipse.reddeer.core/META-INF/MANIFEST.MF]: Yes, if you want to test SWT app (no Workbench)

File: plugins/org.eclipse.reddeer.core/src/org/eclipse/reddeer/core/lookup/WidgetLookup.java (hunk 2 above)
Comment (rawagner) [reply plugins/org.eclipse.reddeer.core/src/org/eclipse/reddeer/core/lookup/WidgetLookup.java]: do you mean getWorkbenchShell()/getShellForActiveWorkbench() ? IMHO it is a better naming

**Comment by rawagner** [comment]
testPR

File: plugins/org.eclipse.reddeer.e4.workbench/src/org/eclipse/reddeer/workbench/impl/part/AbstractWorkbenchPart.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc and others.
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.e4.workbench/src/org/eclipse/reddeer/workbench/impl/part/AbstractWorkbenchPart.java]: There should be 2018

Comment (rawagner) [reply plugins/org.eclipse.reddeer.e4.workbench/src/org/eclipse/reddeer/workbench/impl/part/AbstractWorkbenchPart.java]: actually the files were created in 2017 and there is no need to change the date for the updated files. A date in license header represents a year when the file was created

**Pull request merged**